### PR TITLE
Switch assertions test library to AssertJ 

### PR DESCRIPTION
**Pull request by mslowiak**
Closes: #123

### CODE REVIEW

**Comment by mfvanek** [comment]
Hi @mslowiak,
Thank you for this PR.

There is an issue with using secrets in CI workflow from forks. I'm working on it.
https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
https://github.community/t/make-secrets-available-to-builds-of-forks/16166/37
https://github.community/t/allow-secrets-to-be-shared-with-forks-from-trusted-actions/16525

**Comment by mslowiak** [comment]
> Hi @mslowiak,
> 
> Thank you for this PR.
> 
> 
> 
> There is an issue with using secrets in CI workflow from forks. I'm working on it.
> 
> https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
> 
> https://github.community/t/make-secrets-available-to-builds-of-forks/16166/37
> 
> https://github.community/t/allow-secrets-to-be-shared-with-forks-from-trusted-actions/16525

Take your time. If you will need any input from my side let me know

**Comment by mfvanek** [comment]
@mslowiak 
There are failed tests
```
> Task :test

PgParamImplTest > equalsAndHashCode() FAILED
    org.opentest4j.AssertionFailedError at PgParamImplTest.java:84

ConfigurationMaintenanceOnHostImplTest > getCurrentValueForUnknownParam() FAILED
    org.opentest4j.AssertionFailedError at ConfigurationMaintenanceOnHostImplTest.java:86

PgConnectionImplTest > equalsAndHashCode() FAILED
    org.opentest4j.AssertionFailedError at PgConnectionImplTest.java:78

PrimaryHostDeterminerImplTest > isPrimaryWithExecutionError() FAILED
    org.opentest4j.AssertionFailedError at PrimaryHostDeterminerImplTest.java:59

QueryExecutorTest > executeQueryWithBloatThresholdWithExecutionError() FAILED
    org.opentest4j.AssertionFailedError at QueryExecutorTest.java:98

QueryExecutorTest > executeQueryWithSchemaWithExecutionError() FAILED
    org.opentest4j.AssertionFailedError at QueryExecutorTest.java:82

ForeignKeyTest > shouldCreateDefensiveCopyOfColumnsList() FAILED
    java.lang.AssertionError at ForeignKeyTest.java:55
```

See https://github.com/mfvanek/pg-index-health/runs/5677886375?check_suite_focus=true

**Review by mfvanek — CHANGES_REQUESTED** [review]
Need to fix failed tests

**Comment by mslowiak** [comment]
@mfvanek
I fixed the problematic tests. I used incorrect assertion methods / refactored in a bad way.
You can recheck it

File: src/test/java/io/github/mfvanek/pg/common/health/DatabaseHealthImplTest.java [hunk 1]
```diff
@@ -63,480 +55,440 @@
...
-                    final List<Index> invalidIndexes = databaseHealth.getInvalidIndexes(ctx);
-                    assertNotNull(invalidIndexes);
-                    assertThat(invalidIndexes, empty());
-                });
+        executeTestOnDatabase(schemaName, DatabasePopulator::withReferences, ctx -> {
+            final List<Index> invalidIndexes = databaseHealth.getInvalidIndexes(ctx);
+            assertThat(invalidIndexes).isNotNull();
+            assertThat(invalidIndexes).isEmpty();
+        })
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/common/health/DatabaseHealthImplTest.java:70]: Please remove the hanging semicolons in this file

File: src/test/java/io/github/mfvanek/pg/common/health/logger/ExclusionsBuilderTest.java [hunk 2]
```diff
@@ -12,20 +12,18 @@
...
-                        "indexBloatSizeThresholdInBytes=0, indexBloatPercentageThreshold=0, " +
-                        "tableBloatSizeThresholdInBytes=0, tableBloatPercentageThreshold=0}",
-                builder.toString());
+        assertThat(builder).isNotNull();
+        assertThat(builder.toString()).isEqualTo("ExclusionsBuilder{duplicatedIndexesExclusions='', intersectedIndexesExclusions='', " +
+                "unusedIndexesExclusions='', tablesWithMissingIndexesExclusions='', tablesWithoutPrimaryKeyExclusions='', " +
+                "indexesWithNullValuesExclusions='', indexSizeThresholdInBytes=0, tableSizeThresholdInBytes=0, " + "indexBloatSizeThresholdInBytes=0, indexBloatPercentageThreshold=0, " +
+                "tableBloatSizeThresholdInBytes=0, tableBloatPercentageThreshold=0}");
     }
+
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/common/health/logger/ExclusionsBuilderTest.java:28]: Please remove empty line

File: src/test/java/io/github/mfvanek/pg/common/health/logger/HealthLoggerTest.java [hunk 3]
```diff
@@ -52,34 +50,27 @@
...
                     assertContainsKey(logs, SimpleLoggingKey.INVALID_INDEXES, "invalid_indexes\t1");
                     assertContainsKey(logs, SimpleLoggingKey.DUPLICATED_INDEXES, "duplicated_indexes\t2");
                     assertContainsKey(logs, SimpleLoggingKey.FOREIGN_KEYS, "foreign_keys_without_index\t1");
                     assertContainsKey(logs, SimpleLoggingKey.TABLES_WITHOUT_PK, "tables_without_primary_key\t1");
                     assertContainsKey(logs, SimpleLoggingKey.INDEXES_WITH_NULLS, "indexes_with_null_values\t1");
                     assertContainsKey(logs, SimpleLoggingKey.INDEXES_BLOAT, "indexes_bloat\t11");
                     assertContainsKey(logs, SimpleLoggingKey.TABLES_BLOAT, "tables_bloat\t2");
-                });
+                })
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/common/health/logger/HealthLoggerTest.java:66]: Please remove the hanging semicolons

File: src/test/java/io/github/mfvanek/pg/common/management/DatabaseManagementImplTest.java [hunk 4]
```diff
@@ -56,63 +49,43 @@
...
+            assertThat(getSeqScansForAccounts(ctx)).isGreaterThanOrEqualTo(AMOUNT_OF_TRIES);
+            databaseManagement.resetStatistics();
+            waitForStatisticsCollector();
+            assertThat(getSeqScansForAccounts(ctx)).isEqualTo(0L);
+            final Optional<OffsetDateTime> statsResetTimestamp = databaseManagement.getLastStatsResetTimestamp();
+            assertThat(statsResetTimestamp).isNotNull();
+            assertThat(statsResetTimestamp).isPresent();
+            assertThat(statsResetTimestamp.get()).isAfter(testStartTime);
+        })
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/common/management/DatabaseManagementImplTest.java:64]: Please remove the hanging semicolons

File: src/test/java/io/github/mfvanek/pg/common/management/DatabaseManagementImplTest.java [hunk 5]
```diff
@@ -56,63 +49,43 @@
...
-                    final Set<PgParam> paramsCurrentValues = databaseManagement.getParamsCurrentValues();
-                    assertNotNull(paramsCurrentValues);
-                    assertThat(paramsCurrentValues.size(), greaterThan(ImportantParam.values().length));
-                });
+        executeTestOnDatabase(schemaName, dbp -> dbp.withReferences().withData(), ctx -> {
+            final Set<PgParam> paramsCurrentValues = databaseManagement.getParamsCurrentValues();
+            assertThat(paramsCurrentValues).isNotNull();
+            assertThat(paramsCurrentValues.size()).isGreaterThan(ImportantParam.values().length);
+        })
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/common/management/DatabaseManagementImplTest.java:89]: Please remove the hanging semicolons

File: src/test/java/io/github/mfvanek/pg/connection/ConnectionCredentialsTest.java [hunk 6]
```diff
@@ -70,29 +65,29 @@ void testEqualsAndHashCode() {
         final ConnectionCredentials fourth = ConnectionCredentials.ofUrl(DEFAULT_URL, "u", "pswd");
 
         // null
-        assertNotEquals(first, null);
+        assertThat(first).isNotNull();
         //noinspection AssertBetweenInconvertibleTypes
-        assertNotEquals(first, BigDecimal.ZERO);
+        assertThat(BigDecimal.ZERO).isNotEqualTo(first);
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/connection/ConnectionCredentialsTest.java:70]: Incorrect order

File: src/test/java/io/github/mfvanek/pg/connection/ConnectionCredentialsTest.java [hunk 7]
```diff
@@ -104,31 +99,32 @@ void equalsHashCodeShouldAdhereContracts() {
     @Test
     void testToString() {
         ConnectionCredentials credentials = ConnectionCredentials.ofUrl(DEFAULT_URL, "user", "pswrd");
-        assertEquals("ConnectionCredentials{connectionUrls=[jdbc:postgresql://localhost/postgres], userName='user', password='pswrd'}",
-                credentials.toString());
+        assertThat(credentials.toString()).isEqualTo("ConnectionCredentials{connectionUrls=[jdbc:postgresql://localhost/postgres], userName='user', password='pswrd'}")
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/connection/ConnectionCredentialsTest.java:103]: semicolons here

File: src/test/java/io/github/mfvanek/pg/connection/PgConnectionFactoryImplTest.java [hunk 8]
```diff
@@ -23,8 +22,8 @@
     void forUrlTest() {
         final PgConnection pgConnection = pgConnectionFactory.forUrl(
                 "jdbc:postgresql://localhost:5432/postgres?prepareThreshold=0&preparedStatementCacheQueries=0", "postgres", "postgres");
-        assertNotNull(pgConnection);
-        assertEquals("jdbc:postgresql://localhost:5432/postgres?prepareThreshold=0&preparedStatementCacheQueries=0",
-                pgConnection.getHost().getPgUrl());
+        assertThat(pgConnection).isNotNull();
+        assertThat(pgConnection.getHost().getPgUrl()).isEqualTo("jdbc:postgresql://localhost:5432/postgres?prepareThreshold=0&preparedStatementCacheQueries=0");
     }
+
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/connection/PgConnectionFactoryImplTest.java:28]: empty line

File: src/test/java/io/github/mfvanek/pg/connection/PgConnectionImplTest.java [hunk 9]
```diff
@@ -60,26 +55,21 @@ void equalsAndHashCode() {
...
-        assertEquals(first, theSame);
-        assertEquals(first.hashCode(), theSame.hashCode());
+        assertThat(theSame).isEqualTo(first);
+        assertThat(theSame.hashCode()).isEqualTo(first.hashCode());
 
         // others
-        assertNotEquals(first, second);
-        assertNotEquals(first.hashCode(), second.hashCode());
-
-        // another implementation of PgConnection
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/connection/PgConnectionImplTest.java:79]: Please restore deleted code

File: src/test/java/io/github/mfvanek/pg/connection/PgHostImplTest.java [hunk 10]
```diff
@@ -16,79 +16,75 @@
...
 
     @Test
     void toStringTest() {
         final PgHost host = PgHostImpl.ofPrimary();
-        assertNotNull(host);
-        assertEquals("PgHostImpl{pgUrl='jdbc:postgresql://primary', hostNames=[primary], maybePrimary=true}",
-                host.toString());
+        assertThat(host).isNotNull();
+        assertThat(host.toString()).isEqualTo("PgHostImpl{pgUrl='jdbc:postgresql://primary', hostNames=[primary], maybePrimary=true}")
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/connection/PgHostImplTest.java:87]: semicolon

File: src/test/java/io/github/mfvanek/pg/index/maintenance/IndexesMaintenanceOnHostImplTest.java [hunk 11]
```diff
@@ -58,383 +50,348 @@
...
-                    final List<Index> invalidIndexes = indexesMaintenance.getInvalidIndexes(ctx);
-                    assertNotNull(invalidIndexes);
-                    assertThat(invalidIndexes, empty());
-                });
+        executeTestOnDatabase(schemaName, DatabasePopulator::withReferences, ctx -> {
+            final List<Index> invalidIndexes = indexesMaintenance.getInvalidIndexes(ctx);
+            assertThat(invalidIndexes).isNotNull();
+            assertThat(invalidIndexes).isEmpty();
+        })
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/index/maintenance/IndexesMaintenanceOnHostImplTest.java:65]: semicolons in this file

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java [hunk 12]
```diff
@@ -83,48 +78,44 @@ void testToString() {
...
     @Test
     void withoutIndexes() {
-        assertThrows(NullPointerException.class, () -> DuplicatedIndexes.of(null));
-        assertThrows(IllegalArgumentException.class, () -> DuplicatedIndexes.of(Collections.emptyList()));
-        assertThrows(IllegalArgumentException.class, () -> DuplicatedIndexes.of(
-                Collections.singletonList(IndexWithSize.of("t", "i", 1L))));
+        assertThatThrownBy(() -> DuplicatedIndexes.of(null)).isInstanceOf(NullPointerException.class);
+        assertThatThrownBy(() -> DuplicatedIndexes.of(Collections.emptyList())).isInstanceOf(IllegalArgumentException.class);
+        assertThatThrownBy(() -> DuplicatedIndexes.of(Collections.singletonList(IndexWithSize.of("t", "i", 1L)))).isInstanceOf(IllegalArgumentException.class)
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java:92]: semicolon

File: src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java [hunk 13]
```diff
@@ -139,28 +130,31 @@ void testEqualsAndHashCode() {
...
         // the same
-        assertEquals(first, DuplicatedIndexes.of(
-                IndexWithSize.of("t1", "i2", 505L), // different order
-                IndexWithSize.of("t1", "i1", 606L))); // different size
+        assertThat(
+                DuplicatedIndexes.of(// different order
+                        IndexWithSize.of("t1", "i2", 505L), // different size
+                        IndexWithSize.of("t1", "i1", 606L)
+                )
+        ).isEqualTo(first); // different size
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/DuplicatedIndexesTest.java:147]: Please restore original comment positions

File: src/test/java/io/github/mfvanek/pg/model/index/IndexWithBloatTest.java [hunk 14]
```diff
@@ -59,33 +56,33 @@ void equalsAndHashCode() {
         final IndexWithBloat second = IndexWithBloat.of("t2", "i2", 30L, 3L, 10);
         final IndexWithBloat third = IndexWithBloat.of("t3", "i3", 22L, 11L, 50);
 
-        assertNotEquals(first, null);
+        assertThat(first).isNotNull();
         //noinspection AssertBetweenInconvertibleTypes
-        assertNotEquals(first, BigDecimal.ZERO);
+        assertThat(BigDecimal.ZERO).isNotEqualTo(first);
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/IndexWithBloatTest.java:61]: Incorrect order

File: src/test/java/io/github/mfvanek/pg/model/index/IndexWithNullsTest.java [hunk 15]
```diff
@@ -15,39 +15,38 @@
...
+        assertThatThrownBy(() -> IndexWithNulls.of("t", "i", 0, "  ")).isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
     void testToString() {
         final IndexWithNulls index = IndexWithNulls.of("t", "i", 22L, "f");
-        assertEquals("IndexWithNulls{tableName='t', indexName='i', " +
-                "indexSizeInBytes=22, nullableField='f'}", index.toString());
+        assertThat(index.toString()).isEqualTo("IndexWithNulls{tableName='t', indexName='i', " + "indexSizeInBytes=22, nullableField='f'}")
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/index/IndexWithNullsTest.java:49]: semicolon

File: src/test/java/io/github/mfvanek/pg/utils/DatabaseAwareTestBase.java [hunk 16]
```diff
@@ -62,9 +62,10 @@ protected void waitForStatisticsCollector() {
                 // see PGSTAT_STAT_INTERVAL at https://github.com/postgres/postgres/blob/master/src/backend/postmaster/pgstat.c
                 Thread.sleep(500L);
             } catch (InterruptedException e) {
-                fail(e);
+                fail("unknown failure", e);
             }
-        });
+        })
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/utils/DatabaseAwareTestBase.java:68]: semicolon

File: src/test/java/io/github/mfvanek/pg/settings/PgParamImplTest.java [hunk 17]
```diff
@@ -64,26 +61,21 @@ void equalsAndHashCode() {
...
-        assertEquals(first.hashCode(), theSame.hashCode());
+        assertThat(theSame).isEqualTo(first);
+        assertThat(theSame.hashCode()).isEqualTo(first.hashCode());
 
         // others
-        assertNotEquals(first, second);
-        assertNotEquals(first.hashCode(), second.hashCode());
-
-        // another implementation of PgParam
-        final PgParam pgParamMock = Mockito.mock(PgParam.class);
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/settings/PgParamImplTest.java:84]: Please restore deleted code

File: src/test/java/io/github/mfvanek/pg/model/table/TableWithMissingIndexTest.java [hunk 18]
```diff
@@ -15,37 +15,36 @@
...
+        assertThatThrownBy(() -> TableWithMissingIndex.of("t", 0, 0, -1)).isInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
     void testToString() {
         final TableWithMissingIndex table = TableWithMissingIndex.of("t", 11L, 33L, 22L);
-        assertEquals("TableWithMissingIndex{tableName='t', tableSizeInBytes=11, seqScans=33, indexScans=22}",
-                table.toString());
+        assertThat(table.toString()).isEqualTo("TableWithMissingIndex{tableName='t', tableSizeInBytes=11, seqScans=33, indexScans=22}")
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/model/table/TableWithMissingIndexTest.java:47]: semicolon

File: src/test/java/io/github/mfvanek/pg/utils/QueryExecutorTest.java [hunk 19]
```diff
@@ -46,28 +42,28 @@
...
+        assertThatThrownBy(() -> TestUtils.invokePrivateConstructor(QueryExecutor.class)).isInstanceOf(UnsupportedOperationException.class);
     }
 
     @Test
     void executeInvalidQuery() {
         final String invalidSql = "select unknown_field from unknown_table";
-        assertThrows(RuntimeException.class, () -> QueryExecutor.executeQuery(
-                pgConnection, invalidSql, (rs) -> null));
+        assertThatThrownBy(() -> QueryExecutor.executeQuery(pgConnection, invalidSql, (rs) -> null)).isInstanceOf(RuntimeException.class)
+        ;
```
Comment (mfvanek) [inline src/test/java/io/github/mfvanek/pg/utils/QueryExecutorTest.java:52]: semicolons

**Review by mfvanek — CHANGES_REQUESTED** [review]
@mslowiak Great job!
Please process my comments. Most of them are related to formatting

File: src/test/java/io/github/mfvanek/pg/connection/PgConnectionImplTest.java (hunk 9 above)
Comment (mslowiak) [reply src/test/java/io/github/mfvanek/pg/connection/PgConnectionImplTest.java:79]: this cannot be done, as assertion with mock in AssertJ is not supported

File: src/test/java/io/github/mfvanek/pg/settings/PgParamImplTest.java (hunk 17 above)
Comment (mslowiak) [reply src/test/java/io/github/mfvanek/pg/settings/PgParamImplTest.java:84]: this cannot be done, as assertion with mock in AssertJ is not supported

**Comment by mslowiak** [comment]
@mfvanek Fixed

**Review by mfvanek — APPROVED** [review]
Thanks!

**Pull request merged**